### PR TITLE
feat: max daily consultation working as expected

### PR DESCRIPTION
**Pull request by Mjiyako**
This PR implements robust backend validation for student bookings. It ensures that lecturers are not overbooked by enforcing a maximum daily limit of consultations, verified against the MongoDB database.

 Key Changes

Database-Driven Validation: Replaced hardcoded checks with Booking.countDocuments to accurately track existing consultations in real-time.

Daily Consultation Limit: Implemented a check to ensure no lecturer exceeds 10 consultations per day.

Date Integration: Updated the booking middleware and schema to require a date field, ensuring accurate daily tracking.

Testing Suite: Added comprehensive Jest tests in tests/booking.test.js to cover:

Missing required fields (including the new date requirement).

Validation of daily limits (10/10).

Technical Details

The validateLecturerHours middleware now performs two consecutive database queries:

Checks if the lecturer's total count for that date has reached the daily limit.

Updated the Availability model to include dailySessionLimit with a default of 10.

How to Test

Run npm test to verify the new test cases pass.

Try to create a booking via the API without a date field; it should return a 400 error.

Simulate a full day by adding 10 bookings for a lecturer on one date; the 11th booking should be rejected with the new "daily limit" message.

### CODE REVIEW

**Comment by ofentsetembe** [comment]
This is a much-needed upgrade. Moving away from hardcoded values to Booking.countDocuments makes the system significantly more reliable, especially when multiple students are booking simultaneously.

**Comprehensive Testing:** The Jest tests for the 10/10 limit and missing date fields are solid. It's great to see edge cases like "Missing Date" being handled with a 400 status.

**Schema Integrity:** Adding dailySessionLimit to the Availability model is the right move—it keeps the configuration centralized.

Approved!

File: src/app.js [hunk 1]
```diff
@@ -51,7 +51,9 @@ app.post('/api/register', async (req, res) => {
       password: hashedPassword
     })
 
-    await user.save()
+    await user.save();
+    console.log('User registered in DB:', user.email); // This will now show the actual email
```
Comment (ofentsetembe) [inline src/app.js:55]: This is a good thing for when testing and debugging to see if it really stores the correct email. However, we must remember to remove it in the end once we're sure everything works the way it should.

File: src/middleware/booking-validator.js [hunk 2]
```diff
@@ -36,31 +38,50 @@ const validateLecturerHours = async (req, res, next) => {
...
             });
-        } 
-        
-    
-        else if (maxCapacity - currentBookedStudents === 1) {
-            
+        }
+
+        // Warn if this is the last available spot
+        if (maxCapacity - currentBookedStudents === 1) {
```
Comment (ofentsetembe) [inline src/middleware/booking-validator.js:59]: Switched this from an else if to a standalone if to handle the 'last spot' edge case. This allows us to flag a warning to the user while still allowing the booking to proceed successfully.
